django/apps/blue_management/blue_mgnt/views/views.py

```diff
@@ -298,8 +298,13 @@ def login_user(request):
                                  api.info()['brand_identifier']])
 
                 request.session['username'] = username
+                url = urllib.unquote(request.GET.get('next', '/'))
+                is_absolute = bool(urlparse.urlparse(url))
```

Comment:
Hrm, so, we expect next to be a path such as `/some/path`. We want to make sure that we don't redirect in cases like: `http://badurl.com/someattack`. This `is_absolute`, from what I can see, will always be True.

I would expect this code to be something along the lines of:

```
url = urllib.unquote(request.GET.get('next', '/')).strip()
is_absolute = len(url) > 0 && url[0] == '/'

# If the URL has any sort of hostname or is not absolute, then we default to /, it's not what we expect, 
# so we go to the safe assumption
if urlparse.urlparse(url).hostname or not is_absolute:
    url = "/"

from os.path import normpath
# let's make sure we don't have any "/path/../../../../a" weirdness
url = normpath(url)

return redirect(url)
```

Reply:
I've delved a little bit in Python's urlparse module and I captured the sprit of the snippet in a parse/unparse funciton in 0f1f3dc
There was a small catch with the empty path, where normpath defaults to . instead of / but there's a small if statement there.